### PR TITLE
Backend to property managers

### LINKED ISSUE #401: FRONTEND: Connect Property Managers page to backend
**Purpose**
Currently the `/manage/managers` view is using dummy data located in `/src/views/dummyData/` and we should instead be hitting api to get this data from the DB.

**Proposed Solution**
- use axios to hit the `/users/role` api endpoint with a payload of `"userrole": "2"`, where 2 is the enum value for PROPERTY_MANAGER
- use useState to store the result of the API endpoint
- use the reusable `<Modal />` component to display there was an error if the axios call fails
- connect the data stored in useState to the data prop in `<BootstrapTable />` (this may need to be transformed in some way to display data properly)

**Acceptance Criteria**
Please list the requirements the implementation of this feature should meet.
- [ ] View should display a list of property managers, without using the dummyData file
- [ ] View should be tested at some level (maybe this isn't really that important, but would be a bonus!)
- [ ] When clicking a property manager's name, it should successfully take you to the `/manage/managers/:id` page (see screenshot below)
Screen Shot 2020-10-21 at 9 16 02 PM

**Additional context**
- The dummyData exists because back when this view was created the backend didn't provide the right amount of data needed when hitting the `users/role` endpoint

### PR DESCRIPTION
**Pull request by dbridenbeck**
### What issue is this solving?
closes #401 

### Any helpful knowledge/context for the reviewer?
1. I had to resolve merge conflicts over two occasions on this PR because life kept getting in the way for me to finish this PR, so a double check that those were properly merged would probably be wise.

2. Note how I implemented convertManagersDataForTable() on managers.js and how I'm rendering manager.status for the table. Not sure if that's really how we're hoping to implement this, but I gave it my best shot :)

3. Also note that my old laptop broke, I'm on a new one, and having trouble getting the backend up and running, so I'm unable to do any further testing until then, my apologies for the dumpster fire PR here :(

### Please make sure you've attempted to meet the following coding standards
- [x] Code builds successfully
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code
- [x] There aren't any unnecessary commits or files changed (shouldn't touch the /stashed folder unless the issue requests it)

### CODE REVIEW

File: src/views/Settings/index.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ const validationSchema = Yup.object().shape({
 });
 
 const handleFormSubmit = (context, data) => {
+  console.log("data: ", data);
```
Comment (Chris-Boe) [inline src/views/Settings/index.js]: We can take out this console log

File: src/views/Manager/index.js [hunk 2]
```diff
@@ -1,12 +1,11 @@
-import React, { useState } from "react";
+import React, { useContext, useEffect, useState } from "react";
 import ToggleEditTable from "../../components/ToggleEditTable";
 import * as Yup from "yup";
 import { useLocation } from "react-router-dom";
 import * as axios from "axios";
-import { PROPERTY_MANAGER_DATA } from "../pManagerData";
```
Comment (Chris-Boe) [inline src/views/Manager/index.js:6]: Let's delete the ptManagerData file if it's not being used anywhere anymore

File: src/views/managers/managers.js [hunk 3]
```diff
@@ -78,24 +78,58 @@ const selectRow = {
...
-const Managers = () => {
+// transforms data from API into a format that can be used for bootstrap-table-next
+const convertManagersDataForTable = (managersArray) => {
+  const convertedManagers = managersArray.map(manager => {
+    manager.fullName = `${manager.firstName} ${manager.lastName}`;
+    
+    // #####################################################
+    // not exactly sure if this is how we're hoping to implement displaying the "status" field?
+    // I took my best guess with it for now :)
+    if (manager.lastActive || !manager.archived) {
```
Comment (Chris-Boe) [inline src/views/managers/managers.js:86]: I think this is fine for now, but I will follow up with @htharker42 about what the statuses on the mockups actually mean. I also see "Awaiting Approval" but don't know how that differs from "Pending" nor how they get into those states

**Review by Chris-Boe — CHANGES_REQUESTED** [review]
Nice job! Just a console log and unused file to get rid of, as well as some merge conflicts and this is ready to go!

**Comment by darreniqgeo** [comment]
Hey @Chris-Boe thanks again for the careful review! I resolved conflicts and pushed up the changes - I am still working to get the backend up working on my machine so I'm not able to check that everything is working correctly. . . if you have the chance to take a peek and validate things are working, that would be awesome, if not I'll try and get to that by this weekend!

If there are any other issues here please let me know and I'll get going on it this weekend :)

**Review by Chris-Boe — APPROVED** [review]
Thanks for getting this done! I know you're busy so I took the liberty of fixing up a couple small things to get this through.
Couple things on the PR:
 * Remember to still build the frontend so you know it compiles after making changes, there was a bad import that prevented the app from building
* Try to do only rebases and no merging from development or you'll get the ~30 extra duplicate commits from other people on this PR like you see in the list

**Pull request merged**